app/scripts/modules/ecs/src/domain/IEcsLoadBalancer.ts

```diff
@@ -0,0 +1,114 @@
+import { ILoadBalancer, IInstance, IInstanceCounts } from '@spinnaker/core';
+
+export type IListenerActionType =
+  | 'forward'
+  | 'authenticate-oidc'
+  | 'redirect'
+  | 'authenticate-cognito'
+  | 'fixed-response';
+
+export interface IEcsLoadBalancer extends ILoadBalancer {
+  name: string;
+  account: string;
+  availabilityZones: string[];
+  cloudProvider: string;
+  createdTime: number;
+  dnsname: string;
+  listeners: IALBListener[];
+  targetGroups: IEcsTargetGroup[];
+  ipAddressType?: string; // returned from clouddriver
+  deletionProtection: boolean;
+  idleTimeout: number;
+  loadBalancerType: string;
+  securityGroups: string[];
+  subnets?: string[];
+  region: string;
+  vpcId: string;
+}
+
+export interface IEcsLoadBalancerSourceData extends IEcsLoadBalancer {
+  targetGroupServices: { [key: string]: string[] };
+}
+
+export interface IEcsTargetGroup {
+  account: string;
+  attributes?: any;
+  cloudProvider: string;
+  createdTime: number;
+  detachedInstances?: IInstance[];
+  healthCheckProtocol: string;
+  healthCheckPort: number | 'traffic-port';
+  healthCheckPath: string;
+  healthCheckTimeoutSeconds: number;
+  healthCheckIntervalSeconds: number;
+  healthyThresholdCount: number;
+  unhealthyThresholdCount: number;
+  instanceCounts?: IInstanceCounts;
+  instances?: IInstance[];
+  loadBalancerNames: string[];
+  matcher: any;
+  targetGroupArn: string;
+  targetGroupName: string;
+  port: number;
+  protocol: string;
+  provider?: string;
+  region: string;
+  serverGroups?: string[];
+  targetType?: string;
+  type: string; // returned from clouddriver
+  vpcId?: string;
+  vpcName?: string;
+}
+
+export interface IALBListener {
```

Comment:
nice catch! right now we don't do anything differently, and I believe NLB listeners would contain a subset of the fields returned for ALBs (no rules, for example). Since ECS services could use either I'll rename this to be more general.